### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by Zakhrov** [GPUOpen-Effects/FidelityFX-FSR#1]
Any chance we can have samples for Linux using Vulkan?

**Comment by NotStepbro** [comment]
100/10 idea. I need FSR except without the CIA spying on me. Thanks.

**Comment by bkdwt** [comment]
Yeah, gives us Linux support!

**Comment by Rybens92** [comment]
As I heard before, it's working without any issues through Wine/Proton

**Comment by rys** [comment]
We have no plans to add Linux support to our sample framework, but we are fully supportive of anyone that wants to attempt a Cauldron port to non-Windows platforms and will happily answer questions and help out where we can.

**Comment by Hi-Angel** [comment]
> We have no plans to add Linux support to our sample framework, but we are fully supportive of anyone that wants to attempt a Cauldron port to non-Windows platforms and will happily answer questions and help out where we can.

But Cauldron [was ported to Linux](https://github.com/GPUOpen-LibrariesAndSDKs/Cauldron/pull/3). The main problem is that Cauldron itself is unmaintained project. It has some legitimate Windows fixes with no reply since 2019.

So, are you going to support specifically the porting work? Should we ping @Zakhrov who did the port to reopen their PR? I don't see in github any mention of changeset-conflicts, so really the Linux port might be just a matter of merging this PR.

**Comment by Zakhrov** [comment]
> > We have no plans to add Linux support to our sample framework, but we are fully supportive of anyone that wants to attempt a Cauldron port to non-Windows platforms and will happily answer questions and help out where we can.
> 
> But Cauldron [was ported to Linux](https://github.com/GPUOpen-LibrariesAndSDKs/Cauldron/pull/3). The main problem is that Cauldron itself is unmaintained project. It has some legitimate Windows fixes with no reply since 2019.
> 
> So, are you going to support specifically the porting work? Should we ping @Zakhrov who did the port to reopen their PR? I don't see in github any mention of changeset-conflicts, so really the Linux port might be just a matter of merging this PR.

I had attempted to port it back when Cauldron was first out. I've pulled in the latest Cauldron code and will try a fresh attempt.

**Comment by Hi-Angel** [comment]
> > > We have no plans to add Linux support to our sample framework, but we are fully supportive of anyone that wants to attempt a Cauldron port to non-Windows platforms and will happily answer questions and help out where we can.
> > 
> > 
> > But Cauldron [was ported to Linux](https://github.com/GPUOpen-LibrariesAndSDKs/Cauldron/pull/3). The main problem is that Cauldron itself is unmaintained project. It has some legitimate Windows fixes with no reply since 2019.
> > So, are you going to support specifically the porting work? Should we ping @Zakhrov who did the port to reopen their PR? I don't see in github any mention of changeset-conflicts, so really the Linux port might be just a matter of merging this PR.
> 
> I had attempted to port it back when Cauldron was first out. I've pulled in the latest Cauldron code and will try a fresh attempt.

This is cool! However, to make sure that effort won't be spent in vain I'd wait for @rys's to reply first

**Comment by rys** [comment]
First of all, I need to apologise that PRs for Cauldron have been ignored during the project's life and our dev process means it can look abandoned while we develop it internally, rather than develop it out in the open.

It's not our intention to release a project as open source code and then not support it properly when the open source community interacts with it, and we want to do better there with Cauldron and all of the GPUOpen open source projects.

If the Linux port is brought up-to-date with current `HEAD` (v1.4.1) and looks good, and I can add support for building it with our internal CI, then we will work to try and find a good way to bring it into the project. I can't promise anything at this stage.

I'll work with you in my personal spare time on this effort when I can.

**Comment by psi29a** [comment]
> I'll work with you in my personal spare time on this effort when I can. Let's open a new issue against Cauldron and take it from there.

One would hope that AMD would help you carve out a bit of time in the week for you to focus on this. 

Thank you for willing to go above and beyond. :)

**Comment by Zakhrov** [comment]
I've re-opened the pull request and have updated my fork to the latest Cauldron sources. I've fixed the build errors on Linux  but there is still a long way to go

**Comment by rys** [comment]
@psi29a we've discussed it internally a couple of times but there's no project-related justification for non-Windows platform support at the moment.

@Zakhrov great, I'm sure it's a good starting point. Please bear with me when it comes to review + helping get it into the right shape to hopefully try and land it.

**Comment by cb88** [comment]
> we've discussed it internally a couple of times

In the man-hours spend discussing... it could have been implemented twice over typically is how it goes.

**Comment by isc30** [comment]
As a clarification, FSR **is** supported in Linux since it uses standard vulkan GLSL.

What isn't supported at this moment is **building the demo project in linux**, which uses [Cauldron](https://github.com/GPUOpen-LibrariesAndSDKs/Cauldron/tree/3e4a021fc0d3e0372347126771a5155016810a11) that isn't supported there.

Nothing is stopping you from creating a Vulkan App that uses the FSR1 shader in Linux.
Right, @rys ?

**Comment by rys** [comment]
> As a clarification, FSR **is** supported in Linux since it uses standard vulkan GLSL.
> 
> 
> 
> What isn't supported at this moment is **building the demo project in linux**, which uses [Cauldron](https://github.com/GPUOpen-LibrariesAndSDKs/Cauldron/tree/3e4a021fc0d3e0372347126771a5155016810a11) that isn't supported there.
> 
> 
> 
> Nothing is stopping you from creating a Vulkan App that uses the FSR1 shader in Linux.
> 
> Right, @rys ?

Correct, we're discussing Linux support for the Cauldron-based FSR sample, not the core FSR technology.

**Comment by lilianmoraru** [comment]
> @psi29a we've discussed it internally a couple of times but there's no project-related justification for non-Windows platform support at the moment.

Portable code(especially low level libs/tools for a hardware company like AMD) sometimes can open the door to new opportunities, just the things that popped into the head while reading that:
* Support for Linux also means easier support for Android(maybe even helps the partnership with Samsung?), with an easier potential port to other Unix systems: macOS over MoltenVK.
Linux does not mean only desktop/games, it's potential areas for your code(any AMD code) to extend to: car infotainment(helped Nvidia get into cars - which is almost exclusively Linux), kiosks, digital walls, fridges, etc...
* Using Linux tools for quality checks: sanitizers, fuzzers, GCC/Clang warnings
* Some developers could be more productive on Linux with significantly faster I/O(compilation times), idk

**Comment by iMonZ** [comment]
Any news here?

**Comment by linckosz** [comment]
It has been 3 years since the last message, wondering if there is a way to support FSR1 in Linux.
